pkg/ccl/logictestccl/tests/local/local_test: TestCCLLogic_as_of failed

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://mesolite.cluster.engflow.com/invocations/default/44d2b60d-360e-4f10-aa88-b00089ce2c77?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [452501a49d6d1e45df61b14846ee6fa0f92df479](https://github.com/cockroachdb/cockroach/commits/452501a49d6d1e45df61b14846ee6fa0f92df479):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestCCLLogic_as_of11333181
    test_log_scope.go:82: use -show-logs to present logs inline
[13:04:02] setting distsql_workmem='53004B';
[13:04:04] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of: 7 statements
    logic.go:2863: 
         
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:104: 
        expected:
        pq: database "test" does not exist
        got:
        pq: relation "t" does not exist
[13:04:06] --- done: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 29 tests, 1 failures
    logic.go:4657: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:107: error while processing
    logic.go:4657: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:107: too many errors encountered, skipping the rest of the input
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of11333181
--- FAIL: TestCCLLogic_as_of (9.98s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61731

--- a/pkg/ccl/logictestccl/testdata/logic_test/as_of
+++ b/pkg/ccl/logictestccl/testdata/logic_test/as_of
@@ -19,6 +19,14 @@ statement ok
 INSERT INTO t VALUES (2);
 INSERT INTO t2 VALUES (2);
 
+# Increase the closed timestamp target duration so that follower_read_timestamp()
+# goes far enough back in time to reliably predate cluster bootstrap, even in
+# slow (race) builds. The default 3s target yields follower_read_timestamp() =
+# now()-4.2s, which in race builds can land mid-bootstrap (after database
+# creation but before table creation), producing flaky error messages.
+statement ok
+SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1m'
+
 statement error pgcode 3D000 pq: database "test" does not exist
 SELECT * FROM t AS OF SYSTEM TIME follower_read_timestamp()
 
@@ -124,6 +132,11 @@ RESET autocommit_before_ddl
 
 statement ok
 SET DEFAULT_TRANSACTION_USE_FOLLOWER_READS TO FALSE
+
+# Reset the closed timestamp target duration now that follower read tests are
+# done, so the bounded staleness tests below work correctly.
+statement ok
+SET CLUSTER SETTING kv.closed_timestamp.target_duration = '3s'
 
 query B
 SELECT with_min_timestamp(statement_timestamp()) = statement_timestamp()

